geo/src/algorithm/triangulate_delaunay.rs

```diff
@@ -0,0 +1,642 @@
+    ///     (x: 2., y: 4.),
+    ///     (x: 2., y: 0.),
+    ///     (x: 3., y: 2.),
+    ///     (x: 4., y: 0.),
+    /// ];
+    ///
+    /// let tri_force = points.delaunay_triangulation().unwrap();
+    ///
+    /// assert_eq!(vec![
+    ///     Triangle(
+    ///         coord! {x: 1., y: 2.},
+    ///         coord! {x: 0., y: 0.},
+    ///         coord! {x: 2., y: 0.},
+    ///     ),
+    ///     Triangle(
+    ///         coord! {x: 2., y: 4.},
+    ///         coord! {x: 1., y: 2.},
+    ///         coord! {x: 3., y: 2.},
+    ///     ),
+    ///     Triangle(
+    ///         coord! {x: 1., y: 2.},
+    ///         coord! {x: 2., y: 0.},
+    ///         coord! {x: 3., y: 2.},
+    ///     ),
+    ///     Triangle(
+    ///         coord! {x: 3., y: 2.},
+    ///         coord! {x: 2., y: 0.},
+    ///         coord! {x: 4., y: 0.},
+    ///     )],
+    ///     tri_force
+    /// );
+    ///
+    /// ```
+    ///
+    fn delaunay_triangulation(&self) -> Result<Vec<Triangle<T>>>;
+}
+
+impl<T: GeoFloat> TriangulateDelaunay<T> for Polygon<T>
+where
+    f64: From<T>,
+{
+    fn delaunay_triangulation(&self) -> Result<Vec<Triangle<T>>> {
+        let super_triangle = DelaunayTriangle(create_super_triangle(self)?);
+
+        let mut triangles: Vec<DelaunayTriangle<T>> = vec![super_triangle.clone()];
+        for pt in self.exterior().coords() {
+            add_point(&mut triangles, pt)?;
+        }
+        remove_super_triangle(&mut triangles, &super_triangle);
+        Ok(triangles.iter().map(|x| x.0).collect())
+    }
+}
+
+impl<T: GeoFloat> TriangulateDelaunay<T> for MultiPoint<T>
+where
+    f64: From<T>,
+{
+    fn delaunay_triangulation(&self) -> Result<Vec<Triangle<T>>> {
+        let poly = Polygon::new(self.into_iter().cloned().collect(), vec![]);
+        poly.delaunay_triangulation()
+    }
+}
+
+fn create_super_triangle<T: GeoFloat>(geometry: &Polygon<T>) -> Result<Triangle<T>>
+where
+    f64: From<T>,
+{
+    let expand_factor = T::from(DEFAULT_SUPER_TRIANGLE_EXPANSION)
+        .ok_or(DelaunayTriangulationError::FailedToConstructSuperTriangle)?;
+    let bounds = geometry
+        .bounding_rect()
+        .ok_or(DelaunayTriangulationError::FailedToConstructSuperTriangle)?;
+    let width = bounds.width() * expand_factor;
+    let height = bounds.height() * expand_factor;
+    let bounds_min = bounds.min();
+    let bounds_max = bounds.max();
+
+    Ok(Triangle(
+        coord! {x: bounds_min.x - width, y: bounds_min.y},
+        coord! {x: bounds_max.x + width, y: bounds_min.y - height},
+        coord! {x: bounds_max.x + width, y: bounds_max.y + height},
+    ))
+}
+
+fn find_line<T: GeoFloat>(line: &Line<T>, lines: &[Line<T>]) -> Option<usize> {
+    for (idx, sample_line) in lines.iter().enumerate() {
+        if line == sample_line {
+            return Some(idx);
+        }
+    }
+    None
+}
+
+fn add_point<T: GeoFloat>(triangles: &mut Vec<DelaunayTriangle<T>>, c: &Coord<T>) -> Result<()>
```

Comment:
`Result<()>` unnecessary? I dont see any return or `?` inside

Reply:
`c` -> `coord`

Reply:
> `c` -> `coord` or `point`

I renamed `add_point` to `add_coordinate`